view/frontend/web/js/view/cart/cardknox-google-pay.js

```diff
@@ -0,0 +1,171 @@
+                        jQuery(".gpay-error").html("").hide();
+                    }, 4000);
+                    reject(err);
+                }
+            });
+        },
+        onProcessPayment: function (paymentResponse) {
+            paymentResponse =  JSON.parse(JSON.stringify(paymentResponse));
+            var xAmount  = paymentResponse.transactionInfo.totalPrice;
+            if (xAmount <= 0) {
+                jQuery(".gpay-error").html("<div> Payment is not authorized. Invalid amount. Amount must be greater than 0 </div>").show();
+                setTimeout(function () { 
+                    jQuery(".gpay-error").html("").hide();
+                }, 4000);
+                throw "Payment is not authorized. Invalid amount. Amount must be greater than 0";
+            } else {
+                var token = btoa(paymentResponse.paymentData.paymentMethodData.tokenizationData.token);
+                return gPay.startPlaceOrder(token, xAmount, paymentResponse);
+            }
+        },
+
+        onPaymentCanceled: function(respCanceled) {
+            jQuery(".gpay-error").html("<div> Payment was canceled </div>").show();
+            setTimeout(function () { 
+                jQuery(".gpay-error").html("").hide();
+            }, 4000);
+        },
+        handleResponse: function (resp) {
+            const respObj = JSON.parse(resp);
+            if (respObj) {
+                if (respObj.xError) {
+                    setTimeout(function () { console.log(`There was a problem with your order (${respObj.xRefNum})!`) }, 500);
+                } else
+                    setTimeout(function () { console.log(`Thank you for your order (${respObj.xRefNum})!`) }, 500);
+            }
+        },
+        getGPEnvironment: function () {
+            return gPayConfig.GPEnvironment ? gPayConfig.GPEnvironment : "TEST";
+        },
+        initGP: function() {
+            return {
+                merchantInfo: this.merchantInfo,
+                buttonOptions: this.buttonOptions,
+                environment: this.getGPEnvironment(),
+                billingParameters: this.billingParams,
+                shippingParameters: {
+                    emailRequired: this.shippingParams.emailRequired,
+                },
+                onGetTransactionInfo: "gpRequest.onGetTransactionInfo",
+                onBeforeProcessPayment: "gpRequest.onBeforeProcessPayment",
+                onProcessPayment: "gpRequest.onProcessPayment",
+                onPaymentCanceled: "gpRequest.onPaymentCanceled",
+                onGPButtonLoaded: "gpRequest.gpButtonLoaded",
+                isDebug: isDebugEnv
+            };
+        },
+        gpButtonLoaded: function(resp) {
+            if (!resp) return;
+            if (resp.status === iStatus.success) {
+                showHide("divGpay", true);
+            } else if (resp.reason) {
+                jQuery(".gpay-error").html("<div>"+resp.reason+"</div>").show();
```

Comment:
@mstein-ck,
No. There is no sensitive data in **resp.reason** and not harmful for the website. It will execute when google pay button not loaded. it will show error message why google pay button is not loaded.